// @volar/shared/out/requests.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RemoveAllRefSugars = exports.UseWorkspaceTsdkChanged = exports.TsVersionChanged = exports.GetClientTarNameCaseRequest = exports.GetClientAttrNameCaseRequest = exports.GetServerNameCasesRequest = exports.ShowReferencesNotification = exports.RestartServerNotification = exports.WriteVirtualFilesRequest = exports.VerifyAllScriptsRequest = exports.ActiveSelectionRequest = exports.DocumentVersionRequest = exports.RefCloseRequest = exports.TagCloseRequest = exports.D3Request = exports.VSCodeContentRequest = exports.PingRequest = void 0;
const rpc = require("vscode-jsonrpc");
var PingRequest;
(function (PingRequest) {
    PingRequest.type = new rpc.RequestType0('volar/ping');
})(PingRequest = exports.PingRequest || (exports.PingRequest = {}));
var VSCodeContentRequest;
(function (VSCodeContentRequest) {
    VSCodeContentRequest.type = new rpc.RequestType('vscode/content');
})(VSCodeContentRequest = exports.VSCodeContentRequest || (exports.VSCodeContentRequest = {}));
var D3Request;
(function (D3Request) {
    D3Request.type = new rpc.RequestType('volar/d3');
})(D3Request = exports.D3Request || (exports.D3Request = {}));
var TagCloseRequest;
(function (TagCloseRequest) {
    TagCloseRequest.type = new rpc.RequestType('html/tag');
})(TagCloseRequest = exports.TagCloseRequest || (exports.TagCloseRequest = {}));
var RefCloseRequest;
(function (RefCloseRequest) {
    RefCloseRequest.type = new rpc.RequestType('volar/ref');
})(RefCloseRequest = exports.RefCloseRequest || (exports.RefCloseRequest = {}));
var DocumentVersionRequest;
(function (DocumentVersionRequest) {
    DocumentVersionRequest.type = new rpc.RequestType('vue/docUpdated');
})(DocumentVersionRequest = exports.DocumentVersionRequest || (exports.DocumentVersionRequest = {}));
var ActiveSelectionRequest;
(function (ActiveSelectionRequest) {
    ActiveSelectionRequest.type = new rpc.RequestType0('vue/activeSelection');
})(ActiveSelectionRequest = exports.ActiveSelectionRequest || (exports.ActiveSelectionRequest = {}));
var VerifyAllScriptsRequest;
(function (VerifyAllScriptsRequest) {
    VerifyAllScriptsRequest.type = new rpc.RequestType('volar.action.verifyAllScripts');
})(VerifyAllScriptsRequest = exports.VerifyAllScriptsRequest || (exports.VerifyAllScriptsRequest = {}));
var WriteVirtualFilesRequest;
(function (WriteVirtualFilesRequest) {
    WriteVirtualFilesRequest.type = new rpc.RequestType('volar.action.writeVirtualFiles');
})(WriteVirtualFilesRequest = exports.WriteVirtualFilesRequest || (exports.WriteVirtualFilesRequest = {}));
var RestartServerNotification;
(function (RestartServerNotification) {
    RestartServerNotification.type = new rpc.NotificationType('volar.action.restartServer');
})(RestartServerNotification = exports.RestartServerNotification || (exports.RestartServerNotification = {}));
var ShowReferencesNotification;
(function (ShowReferencesNotification) {
    ShowReferencesNotification.type = new rpc.NotificationType('vue.findReferences');
})(ShowReferencesNotification = exports.ShowReferencesNotification || (exports.ShowReferencesNotification = {}));
var GetServerNameCasesRequest;
(function (GetServerNameCasesRequest) {
    GetServerNameCasesRequest.type = new rpc.RequestType('volar/getTagNameCaseServer');
})(GetServerNameCasesRequest = exports.GetServerNameCasesRequest || (exports.GetServerNameCasesRequest = {}));
var GetClientAttrNameCaseRequest;
(function (GetClientAttrNameCaseRequest) {
    GetClientAttrNameCaseRequest.type = new rpc.RequestType('volar/getAttrNameCaseClient');
})(GetClientAttrNameCaseRequest = exports.GetClientAttrNameCaseRequest || (exports.GetClientAttrNameCaseRequest = {}));
var GetClientTarNameCaseRequest;
(function (GetClientTarNameCaseRequest) {
    GetClientTarNameCaseRequest.type = new rpc.RequestType('volar/getTagNameCaseClient');
})(GetClientTarNameCaseRequest = exports.GetClientTarNameCaseRequest || (exports.GetClientTarNameCaseRequest = {}));
var TsVersionChanged;
(function (TsVersionChanged) {
    TsVersionChanged.type = new rpc.NotificationType('volar.tsVersionChanged');
})(TsVersionChanged = exports.TsVersionChanged || (exports.TsVersionChanged = {}));
var UseWorkspaceTsdkChanged;
(function (UseWorkspaceTsdkChanged) {
    UseWorkspaceTsdkChanged.type = new rpc.NotificationType('volar.useWorkspaceTsdkChanged');
})(UseWorkspaceTsdkChanged = exports.UseWorkspaceTsdkChanged || (exports.UseWorkspaceTsdkChanged = {}));
var RemoveAllRefSugars;
(function (RemoveAllRefSugars) {
    RemoveAllRefSugars.type = new rpc.NotificationType('volar/removeAllRefSugars');
})(RemoveAllRefSugars = exports.RemoveAllRefSugars || (exports.RemoveAllRefSugars = {}));
